xdsl/dialects/builtin.py

```diff
@@ -183,14 +166,83 @@ def print_parameter(self, printer: Printer) -> None:
     @staticmethod
     def generic_constraint_coercion(args: tuple[Any]) -> AttrConstraint:
         assert len(args) == 1
-        return ArrayOfConstraint(irdl_to_attr_constraint(args[0]))
+        return ArrayOfConstraint(RangeOf(irdl_to_attr_constraint(args[0])))
```

Comment:
I think I'm getting slightly confused what the generic parameter on `ArrayAttr` means? Why is it a tuple